src/main/java/edu/wpi/grip/ui/preview/ContoursSocketPreviewView.java

```diff
@@ -0,0 +1,84 @@
+package edu.wpi.grip.ui.preview;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.grip.core.OutputSocket;
+import edu.wpi.grip.core.events.SocketChangedEvent;
+import edu.wpi.grip.ui.util.ImageConverter;
+import javafx.application.Platform;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.VBox;
+
+import static org.bytedeco.javacpp.opencv_core.*;
+import static org.bytedeco.javacpp.opencv_imgproc.drawContours;
+import static edu.wpi.grip.core.operations.composite.FindContourOperation.Contours;
+
+/**
+ * A preview view for displaying contours.  This view shows each contour as a different-colored outline (so they can be
+ * individually distinguished), as well as a count of the total number of contours found.
+ */
+public class ContoursSocketPreviewView extends SocketPreviewView<Contours> {
+
+    private final ImageConverter imageConverter = new ImageConverter();
+    private final ImageView imageView = new ImageView();
+    private final Label infoLabel = new Label();
+    private final Mat tmp = new Mat();
+
+    private final static Scalar[] CONTOUR_COLORS = new Scalar[]{
+            Scalar.RED,
+            Scalar.YELLOW,
+            Scalar.GREEN,
+            Scalar.CYAN,
+            Scalar.BLUE,
+            Scalar.MAGENTA,
+    };
+
+    /**
+     * @param eventBus The EventBus used by the application
+     * @param socket   An output socket to preview
+     */
+    public ContoursSocketPreviewView(EventBus eventBus, OutputSocket<Contours> socket) {
+        super(eventBus, socket);
+
+        this.setContent(new VBox(this.imageView, this.infoLabel));
+        this.render();
```

Comment:
>  Synchronization ensures that changes made by one thread become visible to other threads in a predictable manner. On some architectures, in the absence of synchronization, different threads may see memory operations appear to have been executed in a different order than they actually were executed.
> [Brian Goetz](http://www.ibm.com/developerworks/library/j-jtp0618/)


> **What is meant by reordering?**
There are a number of cases in which accesses to program variables (object instance fields, class static fields, and array elements) may appear to execute in a different order than was specified by the program. The compiler is free to take liberties with the ordering of instructions in the name of optimization. Processors may execute instructions out of order under certain circumstances. Data may be moved between registers, processor caches, and main memory in different order than specified by the program.
> \- [Jeremy Manson and Brian Goetz](https://www.cs.umd.edu/~pugh/java/memoryModel/jsr-133-faq.html)